bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -1855,6 +1901,32 @@ TEST_F(BcStTest, dstEndterInternalCycleOnFetchReservedPagesNotReplied) {
                                    testState_.maxRequiredBlockId));
 }
 
+class BcStTestParamFixtureRejectReason : public BcStTest, public testing::WithParamInterface<uint16_t> {};
+// Rejection is handled gracefully while getting missing blocks
+TEST_P(BcStTestParamFixtureRejectReason, dstRejectFetchBlocksMsgOnce) {
+  auto reason = GetParam();
+  ASSERT_NFF(initialize());
+  ASSERT_NFF(dstStartRunningAndCollecting());
+  ASSERT_NFF(fakeSrcReplica_->replyAskForCheckpointSummariesMsg());
+
+  ASSERT_NFF(getMissingblocksStage<void>(EMPTY_FUNC, EMPTY_FUNC, true, true, reason));
```

Comment:
repeating above comment: We need to refactor the 2 flags to make this call clearer, It is very unclear the way it is defined and implemented now.